recipes/hdf5/all/conanfile.py

```diff
@@ -96,6 +93,9 @@ def source(self):
         tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
 
     def build(self):
+        if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
+            # While building it runs some executables like H5detect
+            raise ConanInvalidConfiguration("Current recipe doesn't support cross-building (yet)")
```

Comment:
There is a new method `validate_build()` since conan 1.51.0. I guess it's the good place to prevent cross-building, while allowing to consume an existing binary in case of cross-build scenario.